lib/exercism/team.rb

```diff
@@ -56,45 +55,50 @@ def defined_with(options, inviter=nil)
     tags = [options[:slug], options[:name], options[:tags]].join(',')
     self.tags = Tag.create_from_text(tags)
 
-    recruits = User.find_or_create_in_usernames(potential_recruits(options[:usernames])) if options[:usernames]
-    recruits = options[:users] if options[:users]
-
-    if recruits
-      recruits.each do |recruit|
-        TeamMembership.create(user: recruit, team: self, inviter: inviter)
-      end
-    end
+    users = User.find_or_create_in_usernames(potential_members(options[:usernames])) if options[:usernames]
+    users = options[:users] if options[:users]
+    invite(users, inviter)
 
     self
   end
 
-  def recruit(usernames, inviter)
-    recruits = User.find_or_create_in_usernames(potential_recruits(usernames.to_s)) - all_members
+  def invite_with_usernames(usernames, inviter)
+    invite(
+      User.find_or_create_in_usernames(potential_members(usernames)),
+      inviter
+    )
+  end
+
+  def invite(users, inviter)
+    return unless users.present?
+
+    users = Array(users) - all_members
 
-    recruits.each do |recruit|
-      TeamMembership.create(user: recruit, team: self, inviter: inviter)
+    users.each do |user|
+      TeamMembershipInvite.create(user: user, team: self, inviter: inviter, refused: false)
     end
   end
 
+  def accept_invite(username)
```

Comment:
I agree that in most projects the user model tends to become a god model, fortunately it is not the case here. But should we really "lock" the model from changes just because of this? I don't know the answer, but seems like the `Team` model will become huge this way (soon we will have a request feature with similar methods). An alternative is to find a missing abstraction and keep these in another place, can you help me with this? :)